src/Internal/BalanceTx/BalanceTx.purs

```diff
@@ -239,7 +239,7 @@ runBalancer utxos changeAddress =
 
     case newMinFee == minFee of
       true -> do
-        finalizedTransaction <- lift $ finalizeTransaction balancedTx utxos -- TODO: all available?
+        finalizedTransaction <- finalizeTransaction balancedTx utxos -- TODO: all available?
```

Comment:
Let's convert this TODO into an issue and put the link here

Reply:
@klntsky I've pushed a tiny fix for this TODO. Generally, we want to have all utxos provided when evaluating and finalizing a transaction, not just the available ones. 

Consider the following scenario: The user submits two transactions in parallel. The first transaction spends some output containing a reference script. The second one includes the same output as a reference input. In case this output is locked for the second transaction, `finalizeTransaction` will fail with `UtxoLookupFailedFor`, masking the actual problem.